lib/pavilion/schedulers/plugins/flux.py

```diff
@@ -0,0 +1,332 @@
+        """Filter nodes by features. Flux doesn't have features.        
+        """        
+        return None
+
+    def _available(self) -> bool:
+        """Ensure we can import and talk to flux."""
+        self._fexecutor = flux.job.FluxExecutor()
+        
+        # used for resource list, etc.
+        self._flux = flux.Flux()
+        return flux is not None
+
+    def _prepare_job_formatter(self):
+        """
+        A job doesn't have an easy status command - instead we do a job
+        listing with a particular format. See src/cmd/flux-jobs.py#L44
+        in flux-framework/flux-core for more attributes.
+        """
+        jobs_format = (
+            "{id.f58:>12} {username:<8.8} {name:<10.10} {status_abbrev:>2.2} "
+            "{ntasks:>6} {nnodes:>6h} {runtime!F:>8h} {success} {exception.occurred}"
+            "{exception.note} {exception.type} {result} {runtime} {status}"
+            "{ranks:h} {t_remaining} {annotations}"
+        )
+        self.jobs_formatter = flux.job.JobInfoFormat(jobs_format)
+
+        # Note there is no attr for "id", its always returned
+        fields2attrs = {
+            "id.f58": (),
+            "username": ("userid",),
+            "exception.occurred": ("exception_occurred",),
+            "exception.type": ("exception_type",),
+            "exception.note": ("exception_note",),
+            "runtime": ("t_run", "t_cleanup"),
+            "status": ("state", "result"),
+            "status_abbrev": ("state", "result"),
+            "t_remaining": ("expiration", "state", "result"),
+        }
+
+        # Set job attributes we will use later to get job statuses
+        self.job_attrs = set()
+        for field in self.jobs_formatter.fields:
+            if field not in fields2attrs:
+                self.job_attrs.update((field,))
+            else:
+                self.job_attrs.update(fields2attrs[field])
+
+
+    def _kickoff(self, pav_cfg, job: Job, sched_config: dict) -> JobInfo:
+        """Submit the kick off script using sbatch."""
+
+        self._prepare_job_formatter()
+        output = job.sched_log.as_posix()
+        error = job.sched_log.as_posix() + ".err"
+
+        # Generate the flux job
+        # Assume the filename includes a hashbang
+        # flux does not support mem_mb, disk_mb
+        fluxjob = JobspecV1.from_command(
+            command=["/bin/bash", job.kickoff_path.as_posix()],
+            num_tasks=sched_config['tasks_per_node'] or 1,
+        )
+        
+        # A duration of zero (the default) means unlimited
+        fluxjob.duration = sched_config['time_limit'] or 0
+        fluxjob.stdout = output
+        fluxjob.stderr = error
+
+        # This doesn't seem to be used?
+        fluxjob.cwd = str(job.tests_path)
+        fluxjob.environment = dict(os.environ)
+        flux_future = self._fexecutor.submit(fluxjob)
+        
+        job._jobid = str(flux_future.jobid())
+        job._submit_time = time.time()
```

Comment:
Ok, cool, you've realized you need to convert the job info into a serializable format to save to file. The _flux_future probably won't ever get used (once a job is scheduled in the run process, that's all Pavilion does with the job object other than hit `save()`.